vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -687,6 +770,14 @@ def forward(self, hidden_states: torch.Tensor,
             activation=self.activation,
         )
 
+        if self.dp_size > 1:
+            start = 0 if self.dp_rank == 0 else cumsum_tokens_across_dp[
+                self.dp_rank - 1]
+            end = cumsum_tokens_across_dp[self.dp_rank]
+
+            all_hidden_states = get_dp_group().all_reduce(final_hidden_states)
+            final_hidden_states = all_hidden_states[start:end, :]
```

Comment:
this might not be compatible with cudagraph, since `start` depends on other ranks' data that are not considered in cudagraph.

Reply:
You're right, this doesn't work with CUDA graphs. I had them working here [bb4f8ae](https://github.com/vllm-project/vllm/pull/13931/commits/bb4f8aeedde7a8efa81e0a101e9f5614460bfcb3), but this was _very_ slow due to useless communication and compute. 

I think we can overcome the excess compute with some minor modifications to the kernels (and moving `cu_tokens_aross_dp` onto the GPU), but the excess communication is trickier, at least without DeepEP. For now I've taken out CUDA graph support

Reply:
agree cudagraph support would be tricky. let's not consider cudagraph right now then.